src/renderer/components/swap/EditableAddress.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { useState, useMemo, useCallback } from 'react'
+
+import { Address } from '@xchainjs/xchain-client'
+import { Asset } from '@xchainjs/xchain-util'
+import { Form, Input } from 'antd'
+import * as FP from 'fp-ts/lib/function'
+import * as O from 'fp-ts/Option'
+import { useIntl } from 'react-intl'
+
+import { AddressValidation } from '../../services/clients'
+import { InnerForm } from '../shared/form'
+import * as Styled from './EditableAddress.styles'
+
+export type EditableAddressProps = {
+  asset: Asset
+  address: Address
+  onClickOpenAddress: (address: Address) => void
+  onChangeAddress: (address: Address) => void
+  addressValidator: AddressValidation
+}
+export const EditableAddress = ({
+  address,
+  onChangeAddress,
+  onClickOpenAddress,
+  addressValidator
+}: EditableAddressProps) => {
+  const RECIPIENT_FIELD = 'recipient'
+  const intl = useIntl()
+  const [editableAddress, setEditableAddress] = useState<O.Option<Address>>(O.none)
+  const maskedRecipientAddress = useMemo(() => address.substring(0, 7) + '...' + address.slice(-3), [address])
```

Comment:
There is already a component to shorten an address

Reply:
`AddressEllipsis` is not reusable in this case because of the positioning of the icons, since the copy would be the right to the address and then we lose control over the order of the icons.

The problem with `truncateAddress` is that I needs to know about the chain and network, which doesn't seem to me worth passing around through props, since existing solution could do just fine.

Reply:
`truncateAddress` is not as same as maskedRecipientAddress. `truncateAddress` is more "safety" by using a same pattern we used at other places. `chain` is already available in `asset` prop, so just a need to pass `network` down to `EditableAddress`.